### DISCUSSION TITLE
Frontend / backend best practices

### DISCUSSION THREAD

**Discussion by automatedmachine** [vercel/next.js#20732] [Help]
Hi,

Just getting into NextJS, stumbling over from CRA. I was looking at creating a nextjs front end with a restify or fastify back end (obviously REST api), likely deploying on Vercel. Now that I'm learning more about NextJS and how it works, I see that it works perfectly as a back end also. Now I'm wondering about implementation. Can/should I deploy NextJS for both the frontend _and_ the backend, or is it better to use a more traditional NodeJS backend? Or is it better to eliminate the whole front end/back end philosophy since you are using server rendered pages and just use one project for the data handling and the front end all in one?

Thanks in advance for your ideas and feedback!

**Comment by marciomrb** [comment]
Hi, I guess backend vercel don't accept, only the frontend...

**Reply by automatedmachine** [reply]
Hi, thanks for the reply! I don't think that's right though, I read on the front page of nextjs.org: "Optionally create API endpoints to provide backend functionality."

Have a look:
[https://nextjs.org/docs/api-routes/introduction](https://nextjs.org/docs/api-routes/introduction)

**Reply by automatedmachine** [reply]
Sorry I misunderstood at first. Yes you can use the Next JS API routes on Vercel (it is the same company, after all), they run as serverless functions on the Vercel platform.

[https://nextjs.org/docs/deployment#optimized-for-nextjs](https://nextjs.org/docs/deployment#optimized-for-nextjs)
